x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/get_filter.ts

```diff
@@ -66,12 +77,19 @@ export const getFilter = async ({
   const getQueryFilter = loadFields
     ? getQueryFilterLoadFields(services.dataViews)
     : getQueryFilterNoLoadFields;
+
+  const dataTiersFilters = ruleTypesSupportingTierFilters.has(type)
+    ? await getDataTierFilter({ uiSettingsClient: services.uiSettingsClient })
+    : [];
+
+  const mergedFilters = [...(filters ? filters : []), ...dataTiersFilters];
```

Comment:
all other places it's just simple merge using spread operator

```js
[...array1, ...array2]
```

No additional logic involved, so I think introducing a separate function that does not do anything apart from spread merge is redundant. 
And in fact, would make reading code harder. Since, you'll need to go that function to see what exactly it doing